api/gateway/gateway.go

```diff
@@ -67,11 +67,11 @@ type Gateway struct {
 }
 
 // New returns a new instance of the Gateway.
-func New(ctx context.Context, opts ...Option) (*Gateway, error) {
+func New(ctx context.Context, router *mux.Router, opts ...Option) (*Gateway, error) {
 	g := &Gateway{
 		ctx: ctx,
 		cfg: &config{
-			router: mux.NewRouter(),
+			router: router,
```

Comment:
Could we make this an `Option`? Then after all for loop that applies the Options runs, we can check if the router field is nil, and if so give it a default value of `mux.NewRouter`. That way tests don't need to specify it when they don't want to customize the router.

There could be Options that want to rely on the router, which would make ordering tricky, but it's probably best for Options to not reach down into other fields anyway, since that creates fragile ordering assumptions. So Options like that would have to make sure to specify a router and do any route registration or other changes to the router before passing it as an Option to New.